include/albatross/src/models/sparse_gp.hpp

```diff
@@ -610,6 +610,14 @@ class SparseGaussianProcessRegression
     return inducing_point_strategy_;
   }
 
+  GrouperFunction get_grouper_function() const {
+    return independent_group_function_;
+  }
+
+  void set_grouper_function(GrouperFunction &&independent_grouper_function) {
+    independent_group_function_ = independent_grouper_function;
```

Comment:
In this context, where `GrouperFunction` is a template type, `independent_grouper_function` is a [forwarding reference](https://lemire.me/blog/2024/05/13/forwarding-references-in-c/), so you can get the compiler to automatically do the right thing (move or copy) for the context by doing

    independent_group_function_ = std::forward<GrouperFunction>(independent_grouper_function);